controller/array_action/array_mediator_ds8k.py

```diff
@@ -280,6 +278,9 @@ def _create_api_volume(self, name, size_in_bytes, array_space_efficiency, pool_i
     def create_volume(self, name, size_in_bytes, space_efficiency, pool):
         array_space_efficiency = get_array_space_efficiency(space_efficiency)
         api_volume = self._create_api_volume(name, size_in_bytes, array_space_efficiency, pool)
+        logger.debug("updating volume {} in cache".format(api_volume.name))
+        with volume_cache_lock:
+            volume_cache.update({api_volume.name: api_volume.id})
```

Comment:
isn't this the same as:
```suggestion
            volume_cache[api_volume.name] = api_volume.id
```
?

Reply:
done

Reply:
I think I [still see it](https://github.com/IBM/ibm-block-csi-driver/blob/e1129dd15eb66b6d2f7303006161283234c77e84/controller/tests/array_action/ds8k/test_array_mediator_ds8k.py#L137-L138)

Reply:
I don't think we need to assert in the beginning of a test.
but if you want control over the cache in the test, can mock it

Reply:
it is needed for later, If I change it we will get dups.. 
the test is to compare the before and after method. I don't see the problem

Reply:
I don't see how the assert (before calling the tested method) tests the method.
what happens later? what will get dups?

Reply:
its showing that the cache has been actually updated and not been at the same state as before the method call.
dup for update and assert.

Reply:
done